gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -99,11 +101,7 @@ struct {
 	__type(value, u64);
 } ipv6_count SEC(".maps");
 
-struct {
-	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
-	__uint(key_size, sizeof(u32));
-	__uint(value_size, sizeof(u32));
-} events SEC(".maps");
+GADGET_TRACER_MAP(events, 1024 * 256);
```

Comment:
`gadget_submit_buf` uses CORE function `bpf_core_enum_value_exists` to detect at run-time if it needs to use a perf ring buffer or a bpf ring buffer.

However, GADGET_TRACER_MAP unconditionally defines a `BPF_MAP_TYPE_RINGBUF`. How does the fallback to perf ring buffer work then?